circuits/builder/rotate.rs

```diff
@@ -145,12 +150,12 @@ impl<L: PlonkParameters<D>, const D: usize> RotateMethods for CircuitBuilder<L,
         // corresponding to an authority set change event in the epoch end header.
         let mut cursor = *start_position;
 
-        // Get the subarray of the header bytes that we want to verify. The header_hash is used as
-        // the seed for randomness.
-        let prefix_subarray = self.get_fixed_subarray::<MAX_HEADER_SIZE, MAX_PREFIX_LENGTH>(
+        // Get the subarray of the header bytes that we want to verify.
+        let prefix_subarray = self.get_fixed_subarray_unsafe::<MAX_HEADER_SIZE, MAX_PREFIX_LENGTH>(
```

Comment:
For all of these instead of passing in the `seed`, why don't you just pass in the `header_hash` as was done previously and then just construct the seed by concatenating the header_hash with the answer array? and then passing them into this function? otherwise the `seed` is constructed way ahead of where it is used which imo is kind of error-prone and confusing?

Reply:
Yeah I agree, will modify

Reply:
Resolved!